src/lib/expression/expression_factory.hpp

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include <memory>
+
+#include "abstract_expression.hpp"
+#include "array_expression.hpp"
+#include "aggregate_expression.hpp"
+#include "arithmetic_expression.hpp"
+#include "extract_expression.hpp"
+#include "between_expression.hpp"
+#include "binary_predicate_expression.hpp"
+#include "case_expression.hpp"
+#include "external_expression.hpp"
+#include "in_expression.hpp"
+#include "logical_expression.hpp"
+#include "lqp_column_expression.hpp"
+#include "value_expression.hpp"
+#include "value_placeholder_expression.hpp"
+
+/**
+ * This file provides convenience methods to create (nested) Expression objects with little boilerplate
+ *
+ * So this
+ *     const auto value_123 = std::make_shared<ValueExpression>(123);
+ *     const auto value_1234 = std::make_shared<ValueExpression>(1234);
+ *     const auto a_eq_123 = std::make_shared<BinaryPredicateExpression>(PredicateCondition::Equals, int_float_a_expression, value_123);
+ *     const auto a_eq_1234 = std::make_shared<BinaryPredicateExpression>(PredicateCondition::Equals, int_float_a_expression, value_1234);
+ *     const auto null_value = std::make_shared<ValueExpression>(NullValue{});
+ *     const auto case_a_eq_1234 = std::make_shared<CaseExpression>(a_eq_1234, int_float_a_expression, null_value);
+ *     const auto case_a_eq_123 = std::make_shared<CaseExpression>(a_eq_123, int_float_b_expression, case_a_eq_1234);
+ *
+ *  becomes
+ *      case_(equals(a, 123),
+ *            b,
+ *            case_(equals(a, 1234),
+ *                  a,
+ *                  null()))
+ *
+ * Intended for tests, etc.
```

Comment:
What is 'etc'? This might be convenient for you, but my first feeling is that there is too much magic happening here for it to be in the main code.